components/org.wso2.carbon.identity.incommon.mgt/src/main/java/org/wso2/carbon/identity/incommon/mgt/validator/SignatureValidator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.incommon.mgt.validator;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.util.Base64;
+
+public class SignatureValidator {
+
+    private static Log log = LogFactory.getLog(SignatureValidator.class);
+
+    private static final String SIGNATURE_ALGORITHM = "SHA256withRSA";
+
+    public boolean isSignatureVerified(String incommonSignatureValue) {
+
+        boolean isVerified = false;
+        try {
+
+            File file = new File("/home/sahan/Documents/WSO2IS/new2-5.8-m6/wso2is-5.8"
```

Comment:
Fix this file path to a relative path where you store the `incommon-metadata.xml` inside the server.